### ISSUE TITLE
restore exited with code -532462766.

### ISSUE THREAD

**Issue by dosluke** [malforge/mdk2#40]
this happens when attempting to build:

The command ""C:\Users\me\.nuget\packages\mal.mdk2.pbpackager\2.0.6\build\..\tools\mdk.exe" restore "C:\Users\me\source\repos\NewFancyFlightInfo\Mdk.NewFancyFlightInfo.csproj" -interactive" exited with code -532462766.

1>Unhandled exception. System.InvalidOperationException: Sequence contains no elements
1>   at System.Linq.ThrowHelper.ThrowNoElementsException()
1>   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
1>   at Mdk.CommandLine.MsBuild.Install(IConsole console)
1>   at Mdk.CommandLine.MdkProject.LoadAsync(String fileName, IConsole console)+MoveNext()
1>   at Mdk.CommandLine.MdkProject.LoadAsync(String fileName, IConsole console)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
1>   at Mdk.CommandLine.Program.RestoreAsync(Parameters parameters, IConsole console, IHttpClient httpClient, IInteraction interaction)
1>   at Mdk.CommandLine.Program.RestoreAsync(Parameters parameters, IConsole console, IHttpClient httpClient, IInteraction interaction)
1>   at Mdk.CommandLine.Program.RunAsync(Parameters parameters, IConsole console, IHttpClient httpClient, IInteraction interaction)
1>   at Mdk.CommandLine.Program.Main(String[] args)
1>   at Mdk.CommandLine.Program.<Main>(String[] args)
1>C:\Users\me\.nuget\packages\mal.mdk2.pbpackager\2.0.6\build\Mal.Mdk2.PbPackager.props(14,9): error MSB3073: The command ""C:\Users\me\.nuget\packages\mal.mdk2.pbpackager\2.0.6\build\..\tools\mdk.exe" restore "C:\Users\me\source\repos\NewFancyFlightInfo\Mdk.NewFancyFlightInfo.csproj" -interactive" exited with code -532462766.
1>Done building project "Mdk.NewFancyFlightInfo.csproj" -- FAILED.

**Comment by malware-dev** [comment]
Please supply an example project that exhibits the failure, I cannot fix what I cannot reproduce. Thanks.

**Comment by dosluke** [comment]
[NewFancyFlightInfo.zip](https://github.com/user-attachments/files/17031702/NewFancyFlightInfo.zip)


I dont have time to make an example project rn which reproduces but this is the current project exhibiting the behavior. Im just bug fixing this script for my own personal use.

windows 10 pro 22H2 19045.4894
VS22 Version 17.12.0 Preview 2.0

**Comment by dosluke** [comment]
Take your time if you have other things to do, i made my own script composer and decent minifier which seems to be working well

**Comment by malware-dev** [comment]
But from the stack trace it seems like something is wrong with your .NET install (or rather, MSBuild)?

**Comment by malware-dev** [comment]
The _bug_ from what I can see, is that I don't even consider the fact that there may be no valid MSBuild installation on the computer.

```cs
﻿using System.Linq;
using Mdk.CommandLine.SharedApi;
using Microsoft.Build.Locator;

namespace Mdk.CommandLine;

/// <summary>
/// MSBuild registration
/// </summary>
public static class MsBuild
{
    /// <summary>
    /// Installs the MSBuild instance available on the system.
    /// </summary>
    /// <param name="console"></param>
    public static void Install(IConsole console)
    {
        if (MSBuildLocator.IsRegistered)
            return;
        
        var msbuildInstances = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(x => x.Version).ToArray();
        foreach (var instance in msbuildInstances)
            console.Trace($"Found MSBuild instance: {instance.Name} {instance.Version}");
        MSBuildLocator.RegisterInstance(msbuildInstances.First());
    }
}
```

However the error is caused by MSBuildLocator (Microsoft code) not finding it at all.

**Comment by dosluke** [comment]
yeah thats an assumption id make too, however there is a valid one because i build and ran a project to make my own composer/minifier with CS... if i can build and run other projects, then there would have to be something off with the MDK?

**Comment by dosluke** [comment]
i see that the code you posted using MS code.. maybe some other configuration thing within MDK?

**Comment by malware-dev** [comment]
Can't be, really. There's no config to set up, and that's the official tool to get MSBuild locations, and it's _that_ which fails.

**Comment by malware-dev** [comment]
And also, why are you the first to report this, if it was a config thing in MDK?

**Comment by dosluke** [comment]
IF it is within MDK it could be because its an obscure bug.. but yeah its def looking like its from MS... but im wondering how my other projects can be built.. but something about MDK causes MS code to fail? not something wrong with MDK, just something MS didnt expect maybe.

**Comment by malware-dev** [comment]
One thing I _can_ think of, is that something has changed and the nuget package must be updated

**Comment by dosluke** [comment]
on your end or mine?

**Comment by malware-dev** [comment]
Mine.

As for why you can build, you haven't mentioned which IDE you're using but at least Visual Studio references msbuild directly and don't really need to use this package.

**Comment by dosluke** [comment]
yeah i mentioned it up top:
windows 10 pro 22H2 19045.4894
VS22 Version 17.12.0 Preview 2.0

**Comment by malware-dev** [comment]
Oh right.

Well... you know.

Preview. That might be the answer right there.

**Comment by dosluke** [comment]
you know i hadnt thought of that.. kind of obvious tho.. ill get the non preview version and see what happens.

**Comment by dosluke** [comment]
yeah looks like thats the problem, my bad, it seemed like it had to be MDK with it only happening in MDK.

Side not, is it possible to make it so the MDK tells me where this error happened in original files, instead of the intermediate script? should i make a separate feature request for this?

![image](https://github.com/user-attachments/assets/7a278757-9e49-4848-bb18-ebad3c0d3301)

**Comment by malware-dev** [comment]
Well you _did_ find a bug, as MDK really should react better if it can't find MSBuild. 

And: If you get errors in the intermediate file, that's almost always an MDK bug. So please make a separate _bug report,_ not a feature request. A workaround for your sake would be to temporarily not use regions (unless this is an MDK region, of course) or otherwise try to move the region statements around to find one that works.

**Comment by dosluke** [comment]
hah! 2 bugs in one go. or 3 if you count the preview version. ill make a second one for this new bug.

**Comment by malware-dev** [comment]
Fixed in v2.0.9
